lib/msf/base/sessions/command_shell.rb

```diff
@@ -88,6 +88,21 @@ def shell_init
     return true
   end
 
+  def bootstrap(datastore = {}, handler = nil)
+    session = self
+
+    if datastore['AutoVerifySession']
+      token = Rex::Text.rand_text_alphanumeric(8..24)
+      response = shell_command("echo #{token}")
```

Comment:
Adding a timeout that's less than 5 seconds I think would make a lot of sense since `echo` isn't going to take a long time to run. If you mean a user-configurable datastore option though, my concern with that is we then get back into an issue where some session types have options that others don't. For example, Meterpreter already dropped the `AutoVerifySession` option, so allowing users to then configure a timeout would be two options unique to shell sessions. The larger goal here is to provide better handling of the different capabilities in session types so I'd like to avoid adding differences.

Reply:
Nope, I mean method-configurable only, particularly for your use case. It would finally have parity with `shell_command_token`:

https://github.com/rapid7/metasploit-framework/blob/bac4ac4aab036e8555efee9c3c8261a2901c3549/lib/msf/core/session/provider/single_command_shell.rb#L82-L89